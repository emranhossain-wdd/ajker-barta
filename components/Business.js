import { useRouter } from 'next/router';
import React, { useEffect, useState } from 'react';
import { IoIosArrowForward } from 'react-icons/io'

const Business = ({business}) => {
 
  const router = useRouter()
  const newses = business.reverse()
  return (
    <div className='mb-6'>
      <div className="container">
        <div className='flex items-center'>
          <h1 onClick={() => router.push('/business')} className='ml-2 text-xl cursor-pointer font-medium text-blue-900'>Business </h1><IoIosArrowForward className='text-red-600 mt-1' />
        </div>
        <div className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-2">
          {
            newses?.slice(0, 4).map(news => <div key={news._id} className=' p-2 shadow-sm hover:shadow-md'>
              <img className='w-full h-52 object-cover' src={news?.images?.img1} alt="" />
              <h1 className='text-lg leading-6 my-1 font-semibold hover:text-red-600 transition-colors duration-300 cursor-pointer' >{news?.heading}</h1>
              <p className='text-sm'>{news?.description[0].slice(0, 100)}...</p>
              <p className='text-md italic text-blue-600'>{news.publishedDate}</p>
            </div>)
          }
        </div>
      </div>
    </div>
  );
};

export default Business;