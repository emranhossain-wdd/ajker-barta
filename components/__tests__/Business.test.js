import React from 'react'
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react'
import Business from '../Business'
import renderer from 'react-test-renderer'
const testArr = [
  {
    _id: "620a0d73b2c17eb44612513b",
    heading: "Scholz in Ukraine to head off invasion threat",
    description: [
      "German Chancellor Olaf Scholz lands in Kyiv on Monday before visiting Moscow to try to head off a “very critical” threat of a Russian invasion that could spark the worst crisis since the Cold War.",
      "The German leader visits the two capitals in reverse order from that taken last week by French president Emmanuel Macron in his bid to quiet the drumbeats of war echoing across eastern Europe.",
      "Russian President Vladimir Putin has surrounded Ukraine from nearly all sides with more than 100,000 soldiers in a high-stakes standoff with the West over NATO’s post-Soviet expansion into countries once under the Kremlin’s domain.",
    ],
    images: {
      img1: "https://images.prothomalo.com/prothomalo-english%2F2022-02%2F4bef49f6-66a4-41a4-b051-6f01977311ed%2F724673_01_02.jpg?auto=format%2Ccompress&fmt=webp&format=webp&w=940&dpr=1.3"
    },
    reporter: "Kyiv",
    subCategory: "europe",
    category: "international",
    publishedDate: "2/14/2022, 2:06:11 PM",
    likes: [],
  },
  {
    _id: "620a0d73b2c17eb44612513b",
    heading: "Scholz in Ukraine to head off invasion threat",
    description: [
      "German Chancellor Olaf Scholz lands in Kyiv on Monday before visiting Moscow to try to head off a “very critical” threat of a Russian invasion that could spark the worst crisis since the Cold War.",
      "The German leader visits the two capitals in reverse order from that taken last week by French president Emmanuel Macron in his bid to quiet the drumbeats of war echoing across eastern Europe.",
      "Russian President Vladimir Putin has surrounded Ukraine from nearly all sides with more than 100,000 soldiers in a high-stakes standoff with the West over NATO’s post-Soviet expansion into countries once under the Kremlin’s domain.",
    ],
    images: {
      img1: "https://images.prothomalo.com/prothomalo-english%2F2022-02%2F4bef49f6-66a4-41a4-b051-6f01977311ed%2F724673_01_02.jpg?auto=format%2Ccompress&fmt=webp&format=webp&w=940&dpr=1.3"
    },
    reporter: "Kyiv",
    subCategory: "europe",
    category: "international",
    publishedDate: "2/14/2022, 2:06:11 PM",
    likes: [],
  }
]
it('render Business component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Business />, div)
})

it('testing business with props', () => {
  const { getByTestId } = render(<Business testArr={testArr} />)
  expect(getByTestId('businessId')).toBeTruthy()
})
it('maching snapshot', () => {
  const tree = renderer.create(<Business />)
  expect(tree).toMatchSnapshot()
})

