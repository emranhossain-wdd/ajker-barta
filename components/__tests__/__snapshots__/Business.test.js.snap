// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="mb-5"
  data-testid="businessId"
>
  <div
    className="container"
  >
    <div
      className="flex items-center"
    >
      <h1
        className="ml-2 text-xl cursor-pointer font-medium text-blue-900"
        onClick={[Function]}
      >
        Business 
      </h1>
      <svg
        className="text-red-600 mt-1"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
        />
      </svg>
    </div>
    <div
      className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-2"
    />
  </div>
</div>
`;
